src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -99,33 +99,35 @@ private async Task TriggerAnalysisAsync(Request request, IIssueConsumer consumer
             await TaskScheduler.Default;
 
             logger.WriteLine($"Analyzing {request.File}");
-            var messages = new List<Message>();
-            Action<Message> handleMessage = message => messages.Add(message); // todo: change to consumer.Accept
+            int issueCount = 0;
+            Action<Message> handleMessage = message => HandleMessage(message, request, consumer, ref issueCount);
 
             // We're tying up a background thread waiting for out-of-process analysis. We could
             // change the process runner so it works asynchronously. Alternatively, we could change the
             // RequestAnalysis method to be asynchronous, rather than fire-and-forget.
             CFamilyHelper.CallClangAnalyzer(handleMessage, request, new ProcessRunner(settings, logger), analysisStatusNotifier, logger, cancellationToken);
 
-            Debug.Assert(messages.All(m => m.Filename == request.File), "Issue for unexpected file returned");
-
-            var issues = messages
-                .Where(m => IsIssueForActiveRule(m, request.RulesConfiguration))
-                .Select(m => CFamilyHelper.ToSonarLintIssue(m, request.CFamilyLanguage, request.RulesConfiguration))
-                .ToList();
-
             telemetryManager.LanguageAnalyzed(request.CFamilyLanguage); // different keys for C and C++
 
-            logger.WriteLine($"Found {issues.Count} issue(s)");
+            logger.WriteLine($"Found {issueCount} issue(s)");
```

Comment:
I think it's worth adding the fileName into the message. There could be multiple analysis running at the same time.